compilation errors with ITK 5.4 and recent boost

```
/home/KrisThielemans/devel/buildConda/sources/pet_rd_tools/lib/nmtools/Common.hpp:252:38: error: cannot convert 'itk::SpatialOrientation::CoordinateTerms' {aka 'itk::SpatialOrientationEnums::CoordinateTerms'} to '__gnu_cxx::__alloc_traits<std::allocator<int>, int>::value_type' {aka 'int'} in assignment
  252 |     coordVals[i] = GetOrientationCode(orient[i]);
```
we need an explicit `static_cast<int>()`, see https://itk.org/Doxygen/html/classitk_1_1SpatialOrientationEnums.html
```
/home/KrisThielemans/devel/buildConda/sources/pet_rd_tools/lib/nmtools/MMR.hpp:612:35: error: 'change_extension' is not a member of 'boost::filesystem'
  612 |       bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
      |                                   ^~~~~~~~~~~~~~~~
/home/KrisThielemans/devel/buildConda/sources/pet_rd_tools/src/NMExtract.cpp: In function 'int main(int, char**)':
/home/KrisThielemans/devel/buildConda/sources/pet_rd_tools/src/NMExtract.cpp:85:27: error: 'complete' is not a member of 'fs'
   85 |   fs::path log_path = fs::complete(fs::current_path());
```
Deprecations:
https://github.com/boostorg/filesystem/blob/41a990ef14760d487c8c3b234f5e3e59178140df/doc/deprecated.html#L244
and `complete` -> `absolute` (from https://www.boost.org/doc/libs/1_83_0/libs/filesystem/doc/deprecated.html)

--- a/lib/nmtools/Common.hpp
+++ b/lib/nmtools/Common.hpp
@@ -27,6 +27,7 @@
 #include <itkImage.h>
 #include <gdcmStringFilter.h>
 #include <exception>
+#include <sstream>
 
 namespace nmtools {
 
@@ -249,7 +250,7 @@ bool SetDesiredCoordinateOrientation(const std::string &target,
 
   //Check they are all valid identifiers
   for (int i = 0; i < 3; i++) {
-    coordVals[i] = GetOrientationCode(orient[i]);
+    coordVals[i] = static_cast<int>(GetOrientationCode(orient[i]));
     if (coordVals[i] == 0){
       LOG(ERROR) << "Unknown coordinate: " << orient[i];
       return false;
@@ -259,9 +260,9 @@ bool SetDesiredCoordinateOrientation(const std::string &target,
   //See itkSpatialOrientation.h
   itk::SpatialOrientation::ValidCoordinateOrientationFlags o =
       (itk::SpatialOrientation::ValidCoordinateOrientationFlags)(
-          ( coordVals[0] << itk::SpatialOrientation::ITK_COORDINATE_PrimaryMinor ) +
-          ( coordVals[1] << itk::SpatialOrientation::ITK_COORDINATE_SecondaryMinor ) +
-          ( coordVals[2] << itk::SpatialOrientation::ITK_COORDINATE_TertiaryMinor ));
+          ( coordVals[0] << static_cast<int>(itk::SpatialOrientation::ITK_COORDINATE_PrimaryMinor )) +
+          ( coordVals[1] << static_cast<int>(itk::SpatialOrientation::ITK_COORDINATE_SecondaryMinor )) +
+          ( coordVals[2] << static_cast<int>(itk::SpatialOrientation::ITK_COORDINATE_TertiaryMinor )));
 
   //Check we don't have an duplicates.
   std::sort(coordVals.begin(), coordVals.end());

--- a/lib/nmtools/MMR.hpp
+++ b/lib/nmtools/MMR.hpp
@@ -274,7 +274,7 @@ FileStatusCode IMMR::CheckForSiemensBFFile(boost::filesystem::path src, uint64_t
   FILE *inFile;
 
   boost::filesystem::path bfPath = src;
-  bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+  bfPath.replace_extension(".bf");
 
   if (!(inFile = fopen(bfPath.string().c_str(), "rb"))) {
       LOG(INFO) << "Cannot open " << bfPath.native();
@@ -363,7 +363,7 @@ bool MMR32BitList::ExtractData( const boost::filesystem::path dst ){
     if ( bfStatus == FileStatusCode::EGOOD ) {
       
       boost::filesystem::path bfPath = _srcPath;
-      bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+      bfPath.replace_extension(".bf").string();
       try {
         if (boost::filesystem::exists(dst)) {
           LOG(ERROR) << "The .bf file already exists!";
@@ -497,12 +497,12 @@ bool MMRSino::ExtractData( const boost::filesystem::path dst ){
   }
 
   boost::filesystem::path bfPath = _srcPath;
-  bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+  bfPath.replace_extension(".bf");
 
   if ( boost::filesystem::exists(bfPath) ){
 
     boost::filesystem::path bfPath = _srcPath;
-    bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+    bfPath.replace_extension(".bf").string();
     try {
       boost::filesystem::copy(bfPath, dst);
       bStatus = true;
@@ -554,7 +554,7 @@ bool MMRSino::IsValid(){
   DLOG(INFO) << "SRC: " << this->_srcPath;
 
   boost::filesystem::path bfPath = _srcPath;
-  bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+  bfPath.replace_extension(".bf").string();
 
   if ( boost::filesystem::exists(bfPath) ){
     LOG(INFO) << ".bf file exists.";
@@ -609,7 +609,7 @@ bool MMRNorm::ExtractData( const boost::filesystem::path dst ){
     if ( bfStatus == FileStatusCode::EGOOD ) {
 
       boost::filesystem::path bfPath = _srcPath;
-      bfPath = boost::filesystem::change_extension(bfPath, ".bf").string();
+      bfPath.replace_extension(".bf").string();
       try {
         boost::filesystem::copy(bfPath, dst);
         bStatus = true;

--- a/lib/nmtools/MRAC.hpp
+++ b/lib/nmtools/MRAC.hpp
@@ -578,7 +578,8 @@ bool MRAC2MU::Write(boost::filesystem::path dst) {
 //Write Interfile header and image pair to disk.
 bool MRAC2MU::WriteToInterFile(boost::filesystem::path dst){
 
-  boost::filesystem::path altPath = boost::filesystem::change_extension(dst, ".mhd");
+  boost::filesystem::path altPath = dst;
+  altPath.replace_extension(".mhd");
   //Write output file
   typedef typename itk::ImageFileWriter<MuMapImageType> WriterType; 
 
@@ -600,7 +601,8 @@ bool MRAC2MU::WriteToInterFile(boost::filesystem::path dst){
 
   std::string outputHeader = GetInterfileHdr();
 
-  altPath = boost::filesystem::change_extension(dst, ".hv");
+  altPath = dst;
+  altPath.replace_extension(".hv");
 
   std::ofstream infoStream;
 

--- a/src/NMExtract.cpp
+++ b/src/NMExtract.cpp
@@ -82,7 +82,7 @@ int main(int argc, char **argv)
   }
 
   //Configure logging
-  fs::path log_path = fs::complete(fs::current_path());
+  fs::path log_path = fs::absolute(fs::current_path());
   log_path /= APP_NAME;
   log_path += "-";
 

--- a/src/NMValidate.cpp
+++ b/src/NMValidate.cpp
@@ -78,7 +78,7 @@ int main(int argc, char **argv)
   }
 
   //Configure logging
-  fs::path log_path = fs::complete(fs::current_path());
+  fs::path log_path = fs::absolute(fs::current_path());
   log_path /= APP_NAME;
   log_path += "-";
 

--- a/src/NMmrac2mu.cpp
+++ b/src/NMmrac2mu.cpp
@@ -81,7 +81,7 @@ int main(int argc, char **argv)
   }
 
   //Configure logging
-  fs::path log_path = fs::complete(fs::current_path());
+  fs::path log_path = fs::absolute(fs::current_path());
   log_path /= APP_NAME;
   log_path += "-";
 

--- a/src/signa.cpp
+++ b/src/signa.cpp
@@ -80,7 +80,7 @@ int main(int argc, char **argv)
   }
 
   //Configure logging
-  fs::path log_path = fs::complete(fs::current_path());
+  fs::path log_path = fs::absolute(fs::current_path());
   log_path /= APP_NAME;
   log_path += "-";
 

